### PR TITLE
Fix release tag for pre-release github action job

### PR DESCRIPTION
**Pull request by BradLugo**
## Description

Fixes release tag for the pre-release GitHub action job

### CODE REVIEW

**Review by tommartensen — CHANGES_REQUESTED** [review]
Thanks for fixing - please review https://github.com/stackrox/stackrox/pull/3734 and merge it into your changes, it fixes finding the versions in the CHANGELOG for minor Y releases. 

Could you provide some context for your changes to the `patch-long-running-cluster` changes? As far as we understood, having `create-long-running-cluster` as a prerequisite to `patch-long-running-cluster` was preferably to repeating the condition. The auth plugin is necessary to avoid a deprecation on GKE/kubectl. 
The NAME should also been fine as it was, since we replace `X.Y.Z-rc.D` with `X-Y-Z-rc-X` with `${NAME//./-}`, and not hard-coding the milestone to the first release candidate would allow to reuse the long-running cluster for the second or third milestone with fewer changes.

**Review by vikin91 — COMMENTED** [review]
I am missing context here - what is this fix about? What is happening without it and what will happen when we have it? How this has been tested?

Without these info, only people who know the code by heart, like @tommartensen & @0x656b694d  can properly review :)

File: .github/workflows/cut-rc.yml [hunk 1]
```diff
@@ -518,14 +518,17 @@ jobs:
...
-    needs: [variables, properties, create-long-running-cluster]
+    needs: [variables, properties]
+    if: >-
+      github.event.inputs.dry-run != 'true' &&
+      needs.variables.outputs.rc != '1' &&
+      github.event.inputs.create-long-cluster != 'false'
     runs-on: ubuntu-latest
     env:
-      NAME: gke-long-running-${{ needs.variables.outputs.milestone }}
+      NAME: gke-long-running-${{ needs.variables.outputs.named-release-patch }}-rc-1
```
Comment (0x656b694d) [inline .github/workflows/cut-rc.yml:528]: Why this change? The code in the steps below replaces periods with dashes in place.

**Comment by BradLugo** [comment]
Sorry y'all, I should've added context to the description. This was more of a temporary fix for the 3.72.2-rc.2 release than it was something to actually get merged in, which might become clear after answering some questions here. Before we get into it, it's important to note that Cut 3.72.2-rc.1 didn't run properly to begin with: Cut 3.72.2-rc.1

> As far as we understood, having create-long-running-cluster as a prerequisite to patch-long-running-cluster was preferably to repeating the condition.
> ...
> The NAME should also been fine as it was, since we replace X.Y.Z-rc.D with X-Y-Z-rc-X with ${NAME//./-}, and not hard-coding the milestone to the first release candidate would allow to reuse the long-running cluster for the second or third milestone with fewer changes.

Some discussion might be needed to resolve this problem, but in the current implementation, `create-long-running-cluster` only runs if `needs.variables.outputs.rc == '1'` (i.e., only on the first release candidate). Since the `create-long-running-cluster` is a prerequisite for `patch-long-running-cluster`, we noticed that `patch-long-running-cluster` wasn't running for 3.72.2-rc.2: https://github.com/stackrox/stackrox/actions/runs/3412566998 (both `patch-long-running-cluster` is skipped here). It might be ideal if there's a way to check if `create-long-running-cluster` was run on any of the named patch versions.

What @house-d and I were discussing yesterday was maybe the ideal solution is to create a long-running cluster named without the release candidate suffix (i.e., just the named release patch:`gke-long-running-${{ needs.variables.outputs.named-release-patch }}`) - perhaps refresh the lifespan after a release candidate is cut. We'd like y'all's input in this as well since y'all definitely know more about this process than we do 🙂 

That said, I'd be willing to believe something obvious is going over my head and that I'm thinking about the long-running cluster workflow all wrong. Please let me know if that's the case.

> The auth plugin is necessary to avoid a deprecation on GKE/kubectl.

When I was trying to gather the evidence to support the change reference here, I couldn't find what I thought was a failed job involving this (I might've incorrectly assumed the job here was relevant, which it doesn't seem actually be: https://github.com/stackrox/stackrox/actions/runs/3388210706/jobs/5643902473). This might not be a problem after all.

---

Hopefully, this explanation helps. I don't think these changes should be merged, but the changes' intention should be applied after we're done with this release. I'm also down to hop in a Slack huddle or meeting to discuss these changes or their intention more. Again, apologizes for not adding context. I should've added this info after we got it working, if not from the very beginning

**Comment by BradLugo** [comment]
@house-d feel free to add anything I missed

**Comment by tommartensen** [comment]
> Some discussion might be needed to resolve this problem, but in the current implementation, create-long-running-cluster only runs if needs.variables.outputs.rc == '1' (i.e., only on the first release candidate). Since the create-long-running-cluster is a prerequisite for patch-long-running-cluster, we noticed that patch-long-running-cluster wasn't running for 3.72.2-rc.2: https://github.com/stackrox/stackrox/actions/runs/3412566998 (both patch-long-running-cluster is skipped here). It might be ideal if there's a way to check if create-long-running-cluster was run on any of the named patch versions.

> What @house-d and I were discussing yesterday was maybe the ideal solution is to create a long-running cluster named without the release candidate suffix (i.e., just the named release patch:gke-long-running-${{ needs.variables.outputs.named-release-patch }}) - perhaps refresh the lifespan after a release candidate is cut. We'd like y'all's input in this as well since y'all definitely know more about this process than we do 🙂

In my understanding, if the provisioning of the long-running cluster (be it in creating or patching phase) failed during RC.1, we should re-run the workflow for RC.1 with manual dispatch. The workflows are idempotent. It is intended to create the long-running cluster only on RC.1. Regarding refreshing the lifespan, I think it might be a good idea to do during cut-rc, especially if a release takes longer than expected. Could you open a new issue in this Epic: https://issues.redhat.com/browse/ROX-10657?

Thanks for your explanations, I will try to root cause the gke-auth-plugin issue.

**Comment by tommartensen** [comment]
> When I was trying to gather the evidence to support the change reference here, I couldn't find what I thought was a failed job involving this (I might've incorrectly assumed the job here was relevant, which it doesn't seem actually be: https://github.com/stackrox/stackrox/actions/runs/3388210706/jobs/5643902473). This might not be a problem after all.

The job you linked uses an outdated cut-rc workflow definition (without the installation of the GKE auth plugin), if you ran it with manual dispatch + selecting workflow from master instead of re-running it on the GHA UI, it would have worked :) 
Therefore, could you please revert the removal of the `USE_GKE_GCLOUD_AUTH_PLUGIN` env var.

**Comment by BradLugo** [comment]
> > Some discussion might be needed to resolve this problem, but in the current implementation, create-long-running-cluster only runs if needs.variables.outputs.rc == '1' (i.e., only on the first release candidate). Since the create-long-running-cluster is a prerequisite for patch-long-running-cluster, we noticed that patch-long-running-cluster wasn't running for 3.72.2-rc.2: [stackrox/stackrox/actions/runs/3412566998](https://github.com/stackrox/stackrox/actions/runs/3412566998) (both patch-long-running-cluster is skipped here). It might be ideal if there's a way to check if create-long-running-cluster was run on any of the named patch versions.
> 
> > What @house-d and I were discussing yesterday was maybe the ideal solution is to create a long-running cluster named without the release candidate suffix (i.e., just the named release patch:gke-long-running-${{ needs.variables.outputs.named-release-patch }}) - perhaps refresh the lifespan after a release candidate is cut. We'd like y'all's input in this as well since y'all definitely know more about this process than we do 🙂
> 
> In my understanding, if the provisioning of the long-running cluster (be it in creating or patching phase) failed during RC.1, we should re-run the workflow for RC.1 with manual dispatch. The workflows are idempotent. It is intended to create the long-running cluster only on RC.1.

Ok, I think I understand. Are you saying that if the Cut 3.72.2-rc.1 workflow had been successful, the `patch-long-running-cluster` job from Cut 3.72.2-rc.2 would have not been skipped?

If the workflows are idempotent, I would expect (perhaps incorrectly) that if `create-long-running-cluster` was successful in the Cut 3.72.2-rc.1 workflow, ' patch-long-running-cluster` could run on any Cut 3.72.2-rc.X workflow. I could also be conflating the idempotency of the _workflow_ with the idempotency of each _job_. My general understanding of idempotency could be totally off too, so in any case, please correct me if I'm wrong. 

---
re: the gke-plugin, yeah I think I was barking up the wrong tree with my changes there. Thanks for taking a look at that 🙂

**Comment by tommartensen** [comment]
@BradLugo This PR is superseded by https://github.com/stackrox/stackrox/pull/3877.

**Pull request closed without merging**